src/vivarium_ihme_obesity_washout/tools/cli.py

```diff
@@ -0,0 +1,69 @@
+from bdb import BdbQuit
+import logging
+from pathlib import Path
+import yaml
+
+import click
+
+from gbd_mapping import causes
+from vivarium_gbd_access.gbd import ARTIFACT_FOLDER
+from vivarium_inputs import get_measure
+from vivarium_inputs.data_artifact import utilities
+from vivarium_inputs.data_artifact.cli import main
+from vivarium_public_health.dataset_manager import Artifact
+
+ASTHMA_DISMOD_ID = 1907  # Abie found from epiviz tool
+
+
+@click.command()
+@click.option('--append', '-a', is_flag=True,
+              help="Preserve existing artifact and append to it")
+@click.option('--verbose', '-v', is_flag=True,
+              help="Turn on debug mode for logging")
+@click.option('--pdb', 'debugger', is_flag=True, help='Drop the debugger if an error occurs')
+def build_washout_artifact(append, verbose, debugger):
+    """
+    build_washout_artifact is a program for building data artifacts locally
+    for the obesity_washout model.
+
+    It will build an artifact for the ``vivarium_ihme_obesity_washout.yaml``
+    model specification file stored in the repository.
+
+    It requires access to the J drive and /ihme. If you are running this job
+    from a qlogin on the cluster, you must specifically request J drive access
+    when you qlogin by adding "-l archive=TRUE" to your command.
+
+    Please have at least 30GB of memory on your qlogin."""
+    model_specification = Path(__file__).parent.parent / 'model_specifications' / 'vivarium_ihme_obesity_washout.yaml'
+    output_root = ARTIFACT_FOLDER / 'vivarium_ihme_obesity_washout'
+
```

Comment:
We should think more about this pattern here.  Esp. w/r/t to locations.  A pattern that Rob introduced that I kind of like involves using template files with jinja.  E.g.:

Template file: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/external_data/yaml_template.in
function to generate concrete specs: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/external_data/build_simulation_files.py
CLI tool to generate model specs: https://github.com/population-interventions/vivarium_unimelb_tobacco_intervention_comparison/blob/master/src/vivarium_unimelb_tobacco_intervention_comparison/cli.py#L30

Definitely not a thing we have to do now, but this approach requires hand editing the yaml files to create artifacts for multiple locs, which is a bit of a drag.  Especially if the underlying model changes (e.g. new causes/risks or alternative models for them).

Reply:
this is interesting. I did something similar for the data requests but it was way more lightweight, just names in brackets and .format(). It worked well, the specs are autogenerated from a template with just different locations and file names